tests/test_transforms/test_math/test_lambda_transform.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from etna.datasets import TSDataset
+from etna.datasets import generate_ar_df
+from etna.transforms import AddConstTransform
+from etna.transforms import LagTransform
+from etna.transforms import LambdaTransform
+from etna.transforms import LogTransform
+
+
+@pytest.fixture
+def ts_non_negative():
+    df = generate_ar_df(
+        start_time="2020-01-01", periods=300, ar_coef=[1], sigma=1, n_segments=3, random_seed=0, freq="D"
+    )
+    df = TSDataset.to_dataset(df)
+    df = df.apply(lambda x: np.abs(x))
+    ts = TSDataset(df, freq="D")
+    return ts
+
+
+@pytest.fixture
+def ts_range_const():
+    periods = 100
+    df_1 = pd.DataFrame(
+        {"timestamp": pd.date_range("2022-06-22", periods=periods), "target": np.arange(0, periods), "segment": 1}
+    )
+    df_2 = pd.DataFrame(
+        {
+            "timestamp": pd.date_range("2022-06-22", periods=periods),
+            "target": np.array([1, 3] * (periods // 2)),
+            "segment": 2,
+        }
+    )
+    df = pd.concat([df_1, df_2])
+    df = TSDataset.to_dataset(df)
+    ts = TSDataset(df, freq="D")
+    return ts
+
+
+@pytest.mark.parametrize(
+    "transform_original, transform_function, out_column",
+    [
+        (
+            LogTransform(in_column="target", out_column="transform_target", inplace=False),
+            lambda x: np.log10(x + 1),
+            "transform_target",
+        ),
+        (
+            AddConstTransform(in_column="target", out_column="transform_target", value=1, inplace=False),
+            lambda x: x + 1,
+            "transform_target",
+        ),
+        (
+            LagTransform(in_column="target", out_column="transform_target", lags=[1]),
+            lambda x: x.shift(1),
+            "transform_target_1",
+        ),
+    ],
+)
+def test_save_transform(ts_non_negative, transform_original, transform_function, out_column):
+    ts_copy = TSDataset(ts_non_negative.to_pandas(), freq="D")
+    ts_copy.fit_transform([transform_original])
+    ts = ts_non_negative
+    ts.fit_transform(
+        [LambdaTransform(in_column="target", out_column=out_column, transform_func=transform_function, inplace=False)]
+    )
+    eps = 1e-9
+    assert set(ts_copy.columns) == set(ts.columns)
+    for column in ts.columns:
+        assert np.all(abs(ts_copy[:, :, column] - ts[:, :, column]) < eps)
```

Comment:
better to use https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html